relay-server/src/actors/project_cache.rs

```diff
@@ -643,39 +642,38 @@ impl Handler<FlushBuckets> for ProjectCache {
         let project_state = match expiry_state {
             ExpiryState::Updated(state) | ExpiryState::Stale(state) => state,
             ExpiryState::Expired => {
+                relay_log::trace!("project expired: merging back {} buckets", buckets.len());
                 // If the state is outdated, we need to wait for an updated state. Put them back into the
-                // aggregator and wait for the next flush cycle.
-                return Box::new(future::err(buckets));
+                // aggregator.
+                Aggregator::from_registry()
+                    .do_send(relay_metrics::MergeBuckets::new(project_key, buckets));
```

Comment:
Please import `MergeBuckets`.

Reply:
The one reason I used fully qualified name here is that the `project_cache` actor already has its own `MergeBuckets` struct / message defined and I wanted to be here perfectly clear which exactly `MergeBuckets` we are using here. 
I would leave it as it is for now and as a part of migrating these actors to `tokio` unify / rename those.

Reply:
yeah, I will make the change in this PR

Reply:
so, I've removed some code duplication from `project_cache` and use the structs from `relay-metrics` crate, see [9e59da1](https://github.com/getsentry/relay/pull/1492/commits/9e59da1d25118e1c2180e96ff2a1531004388430)